account-kit/core/src/store/store.ts

```diff
@@ -62,6 +62,45 @@ export const createAccountKitStore = (
                 return bigintMapReviver(key, value);
               },
             }),
+            merge: (persisted, current) => {
+              const persistedState = persisted as StoreState;
+              const persistedConnections = Array.from(
+                persistedState.connections.values()
+              );
+
+              const connectionsMatch = () =>
+                connections.every((c) => {
+                  const persistedConnection = persistedState.connections.get(
+                    c.chain.id
+                  );
+                  if (!persistedConnection) return false;
+
+                  return Array.from(Object.entries(c)).every(([key, value]) => {
+                    return (
+                      (persistedConnection as Record<string, any>)[key] ===
+                      value
+                    );
+                  });
+                });
+
+              if (
+                // all chains in the passed config should be in persisted state
+                !connections.every((c) =>
+                  persistedState.connections.has(c.chain.id)
+                ) &&
```

Comment:
Should these be ors instead of ands? Also hasn't this condition already been checked in `connectionsMatch` at `if (!persistedConnection) return false`?

Reply:
yup and yup. updating